server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/SnapshotInProgressAllocationDecider.java

```diff
@@ -101,6 +102,18 @@ private static Decision canMove(ShardRouting shardRouting, RoutingAllocation all
                     continue;
                 }
 
+                if (shardSnapshotStatus.state() == SnapshotsInProgress.ShardState.PAUSED_FOR_NODE_REMOVAL) {
+                    // this shard snapshot is paused pending the removal of its assigned node
+                    final var nodeShutdown = allocation.metadata().nodeShutdowns().get(shardRouting.currentNodeId());
+                    if (nodeShutdown != null && nodeShutdown.getType() != SingleNodeShutdownMetadata.Type.RESTART) {
```

Comment:
NIT: could be simplified to:
```
if (allocation.metadata().nodeShutdowns().contains(shardRouting.currentNodeId(), SingleNodeShutdownMetadata.Type.RESTART)) {
```

Reply:
We are looking for a shutdown which is _not_ a `RESTART` here.

Reply:
Missed that